python/lsst/daf/butler/dimensions/_records.py

```diff
@@ -112,9 +112,12 @@ def _createSimpleRecordSubclass(definition: DimensionElement) -> type[SpecificSe
     if definition.spatial:
         members["region"] = (str, ...)
 
-    # mypy does not seem to like create_model
+    # For the new derived class name need to convert to camel case.
+    # so "day_obs" -> "DayObs".
+    derived_name = "".join([part.capitalize() for part in definition.name.split("_")])
```

Comment:
Yes. Most people don't notice since the class is only constructed transiently to validate the JSON content as it's converted to a real DimensionRecord so you only get to see it if your record content fails validation (and physical filter mostly has not changed over the years)